### PR TITLE
UIOR-1307:  handle error message without appropriate affiliation is trying to edit/open PO/POL

### PR DESCRIPTION
**Pull request by alisher-epam**
## Purpose

[UIOR-1307](https://folio-org.atlassian.net/browse/UIOR-1307):  handle error message without appropriate affiliation is trying to edit/open PO/POL
## Approach

https://github.com/user-attachments/assets/1a9b2416-917a-4bfd-8300-e8e5c050b86f

https://github.com/user-attachments/assets/ae7a164f-f55e-481c-952f-3ecf344bd903

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by BogdanDenis — APPROVED** [review]

**Review by Terala-Priyanka — COMMENTED** [review]
Hey @alisher-epam 
Could you please update change log file?
I understand that the lint issues reported are outside your changes. But it would be good to address them.

File: src/common/hooks/useHandleOrderUpdateError/useHandleOrderUpdateError.js [hunk 1]
```diff
@@ -32,6 +37,11 @@ const useHandleOrderUpdateError = (mutatorExpenseClass) => {
             values,
           });
         }
+      } else if (errorMessage?.includes(INVALID_TOKEN_MESSAGE)) {
```
Comment (zburke) [inline src/common/hooks/useHandleOrderUpdateError/useHandleOrderUpdateError.js:40]: Both this condition and the one above handle errors, so it's weird to have one inspect `errorCode` and the other `errorMessage`. Could they be structure the same way?

**Review by zburke — CHANGES_REQUESTED** [review]
* Please update the CHANGELOG
* Please included tests for this new condition you handle

File: src/common/hooks/useHandleOrderUpdateError/useHandleOrderUpdateError.js (hunk 1 above)
Comment (alisher-epam) [reply src/common/hooks/useHandleOrderUpdateError/useHandleOrderUpdateError.js:40]: No, it is not possible. According to the BE team, the error comes from the authorization module and we need to parse the error message to detect the error message.

**Comment by alisher-epam** [comment]
> * Please update the CHANGELOG
> * Please included tests for this new condition you handle

The Changelog has already been added https://github.com/folio-org/ui-orders/blob/master/CHANGELOG.md?plain=1#L34

**Review by BogdanDenis — APPROVED** [review]

**Comment by alisher-epam** [comment]
@folio-org/fe-tl-reviewers ^^ please review

File: src/common/hooks/useHandleOrderUpdateError/useHandleOrderUpdateError.js (hunk 1 above)
Comment (zburke) [reply src/common/hooks/useHandleOrderUpdateError/useHandleOrderUpdateError.js:40]: I understand that top-level error codes like 401 or 403 might be caused by multiple different reasons, but it feels surprising/unfortunate/wrong that our _own_ error codes that we control cannot be more specific. I understand that you are just dealing with the API you've been given, but I would push back against the BE team and ask for a more expressive API.

**Review by usavkov-epam — APPROVED** [review]

File: src/common/hooks/useHandleOrderUpdateError/useHandleOrderUpdateError.js (hunk 1 above)
Comment (alisher-epam) [reply src/common/hooks/useHandleOrderUpdateError/useHandleOrderUpdateError.js]: @zburke updated

**Review by zburke — APPROVED** [review]

**Pull request merged**